### ISSUE TITLE
Questions this plugin

### ISSUE THREAD

**Issue by jenstornell** [getkirby/patterns#7]
I totally missed this plugin. Maybe you have'nt run your release-party just yet?

I have seen your video of it and it looks really interesting (maybe awesome). In fact I tried something like this myself on Kirby 1 but I was not organized, or smart enough to make it work. I did'nt even know what I was trying to build, more than a modular system where CSS and HTML parts was combined as modules. It failed big time.

**I have a few questions (as always):**

1. Is there a reason why files are called `image.html.php` and not just `image.php`?
1. If we have all our code in the patterns, what will be left in the templates and snippets? Not much I guess? Maybe just a reference to include some base pattern?
1. If this works well do you want to merge it to Kirby core or will it always be a plugin?
1. Are you going to take money for it?
1. In what state do you estimate you are in for a well working version? 80%? As good as this thing might be, I hope that Kirby and the panel will get some love as well during this year.

Thanks for your existance!

**Comment by bastianallgeier** [comment]
1. I wanted to make it as clear as possible, what each file contains and keep the option to have more stuff like `image.config.php` etc. 

2. Templates turn mostly in to a set of patterns. You can place your own HTML here and there, but mainly it's a puzzle of patterns. There are some rare use cases for snippets when you use the pattern lab. I keep the header and footer of the HTML document in snippets for example. 

3. I don't think so. One of my main principles for Kirby has always been to not define an opinionated way of how to build things. Forcing every user into this pattern way of building sites, would not work and actually be harmful. 

4. No, I thought about it, but it's going to stay a free extension for Kirby. I want to attract more Kirby users, by providing a better ecosystem of tools around it. That's where my focus is on. 

5. 90% I'm using it in two projects already and it works great. There are some minor issues, which don't really matter much in production. The pattern system is working very reliably, there are only smaller issues with the interface. 

The panel is definitely getting a lot more love this year. Don't worry!! I have lots and lots of plans :)

**Comment by jenstornell** [comment]
Thank you so much for the answers!

3 - Even if you would add Patterns to the Kirby core, it would not force the users to use it. There would still be templates and snippets.

But I agree it's good to have it like a plugin, mainly because it's another UI, like the panel. The panel is a UI for content, Patterns is a UI for code.

4 - Then you probably need to write about it on getkirby.com with screenshots and video. Also a sticky post to the Kirby forum.

5 - Nice! I will try it out in one of my bigger projects later on.

Great news! When I first started with Kirby 1 I did not even had the panel installed. Now I can't live without it.
